src/common/readfield_fi.f90

```diff
@@ -196,12 +196,14 @@ subroutine readfield_fi(istep, nhleft, itimei, ihr1, ihr2, itimefi, ierror)
 
     timepos = iavail(ntav2)%timePos
     timeposm1 = timepos ! Default: No deaccumulation possible
-    if (ntav1 /= 0) then
-      if (iavail(ntav2)%fileNo .eq. iavail(ntav1)%fileNo) then
-        ! previous timestep in same file for deaccumulation
-        timeposm1 = iavail(ntav1)%timePos
-      endif
+    if ((ntav1 /= 0) .and. ((iavail(ntav2)%fileNo .eq. iavail(ntav1)%fileNo))) then
+      ! previous timestep in same file for deaccumulation
+      timeposm1 = iavail(ntav1)%timePos
+    elseif (iavail(ntav2)%pavail_same_file /= 0) then
+      ! previous timestep in same file, even if not in list
```

Comment:
Maybe change to:
`deaccumulation must be from timestep in the same file, even when this timestep is otherwise invalid`

Reply:
slight text change mentioning deaccumulation